modules/prediction/common/message_process.h

```diff
@@ -36,29 +41,44 @@ class MessageProcess {
  public:
   MessageProcess() = delete;
 
-  static bool Init(const PredictionConf &prediction_conf);
+  static bool Init(ContainerManager *container_manager,
+                   EvaluatorManager *evaluator_manager,
+                   PredictorManager *predictor_manager,
+                   const PredictionConf &prediction_conf);
 
-  static bool InitContainers();
+  static bool InitContainers(ContainerManager *container_manager);
 
-  static bool InitEvaluators(const PredictionConf &prediction_conf);
+  static bool InitEvaluators(EvaluatorManager *evaluator_manager,
+                             const PredictionConf &prediction_conf);
 
-  static bool InitPredictors(const PredictionConf &prediction_conf);
+  static bool InitPredictors(PredictorManager *predictor_manager,
+                             const PredictionConf &prediction_conf);
 
   static void ContainerProcess(
+      const std::shared_ptr<ContainerManager> &container_manager,
       const perception::PerceptionObstacles &perception_obstacles);
 
   static void OnPerception(
       const perception::PerceptionObstacles &perception_obstacles,
+      const std::shared_ptr<ContainerManager> &container_manager,
+      EvaluatorManager *evaluator_manager, PredictorManager *predictor_manager,
       PredictionObstacles *const prediction_obstacles);
 
   static void OnLocalization(
+      ContainerManager *container_manager,
```

Comment:
Curious to know why we use raw pointer of ContainerManager in some functions, and  std::shared_ptr<ContainerManager> in others

Reply:
in some function std::shared_ptr of ContainerManager is used to construct an ObstaclesPrioritizer instance, and nothing needs to be constructed by ContainerManager in others.